src/controllers/director.correspondence.address.link.controller.ts

```diff
@@ -0,0 +1,89 @@
+import { NextFunction, Request, Response } from "express";
+import { DIRECTOR_CORRESPONDENCE_ADDRESS_PATH, DIRECTOR_RESIDENTIAL_ADDRESS_PATH } from "../types/page.urls";
+import { Templates } from "../types/template.paths";
+import { urlUtils } from "../utils/url";
+import { createValidationErrorBasic, formatValidationErrors } from '../validation/validation';
+import { OfficerFiling } from "@companieshouse/api-sdk-node/dist/services/officer-filing";
+import { getOfficerFiling, patchOfficerFiling } from "../services/officer.filing.service";
+import { formatTitleCase } from "../services/confirm.company.service";
+import { retrieveDirectorNameFromFiling } from "../utils/format";
+import { DirectorField } from "../model/director.model";
+import { getField } from "../utils/web";
+import { Session } from "@companieshouse/node-session-handler";
+import { SA_TO_ROA_ERROR } from "../utils/constants";
+
+export const get = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const transactionId = urlUtils.getTransactionIdFromRequestParams(req);
+    const submissionId = urlUtils.getSubmissionIdFromRequestParams(req);
+    const session: Session = req.session as Session;
+
+    const officerFiling = await getOfficerFiling(session, transactionId, submissionId);
+
+    return res.render(Templates.DIRECTOR_CORRESPONDENCE_ADDRESS_LINK, {
+      templateName: Templates.DIRECTOR_CORRESPONDENCE_ADDRESS_LINK,
+      backLinkUrl: urlUtils.getUrlToPath(DIRECTOR_CORRESPONDENCE_ADDRESS_PATH, req),
+      directorName: formatTitleCase(retrieveDirectorNameFromFiling(officerFiling)),
+      sa_to_roa: calculateSaToRoaRadioFromFiling(officerFiling.isMailingAddressSameAsRegisteredOfficeAddress),
+    });
+  } catch (e) {
+    return next(e);
+  }
+};
+
+export const post = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const transactionId = urlUtils.getTransactionIdFromRequestParams(req);
+    const submissionId = urlUtils.getSubmissionIdFromRequestParams(req);
+    const session: Session = req.session as Session;
+    const isMailingAddressSameAsRegisteredOfficeAddress = calculateSaToRoaBooleanValue(req);
+
+    if (isMailingAddressSameAsRegisteredOfficeAddress === undefined) {
```

Comment:
I think we should avoid strict checks where possible and just instead do
`if (isMailingAddressSameAsRegisteredOfficeAddress)`
for both, this class and residential implementation

Reply:
What is the reasoning for that? I have based this on the protected details page. It would feel strange to follow that logic here as I see the controller methods always ending with the redirect as the final line. This would mean the error checking would need to form the final part of the method instead and the redirect be nested in the if statement.

Reply:
I had found strict checks leads to bug sneaking across with null values as undefined and null are 2 distinct types with default compiler option strictNullChecks, given this `null !== undefined` but `null == undefined` . 
I see the method `calculateSaToRoaBooleanValue()` strictly only returns a boolean on undefined so I am happy to mark this conversation as solved. 
I added comment as this was my generic response to any strict type checks, I may have missed the protected details page but I'd go with similar comment to that too.